packages/ses/src/intrinsics.js

```diff
@@ -72,68 +73,78 @@ export const makeIntrinsicsCollector = () => {
   const intrinsics = { __proto__: null };
   let pseudoNatives;
 
-  const intrinsicsCollector = {
-    addIntrinsics(newIntrinsics) {
-      initProperties(intrinsics, getOwnPropertyDescriptors(newIntrinsics));
-    },
-
-    // For each intrinsic, if it has a `.prototype` property, use the
-    // whitelist to find out the intrinsic name for that prototype and add it
-    // to the intrinsics.
-    completePrototypes() {
-      for (const [name, intrinsic] of entries(intrinsics)) {
-        if (intrinsic !== Object(intrinsic)) {
-          // eslint-disable-next-line no-continue
-          continue;
-        }
-        if (!objectHasOwnProperty(intrinsic, 'prototype')) {
-          // eslint-disable-next-line no-continue
-          continue;
-        }
-        const permit = whitelist[name];
-        if (typeof permit !== 'object') {
-          throw new Error(`Expected permit object at whitelist.${name}`);
-        }
-        const namePrototype = permit.prototype;
-        if (!namePrototype) {
-          throw new Error(`${name}.prototype property not whitelisted`);
-        }
-        if (
-          typeof namePrototype !== 'string' ||
-          !objectHasOwnProperty(whitelist, namePrototype)
-        ) {
-          throw new Error(`Unrecognized ${name}.prototype whitelist entry`);
-        }
-        const intrinsicPrototype = intrinsic.prototype;
-        if (objectHasOwnProperty(intrinsics, namePrototype)) {
-          if (intrinsics[namePrototype] !== intrinsicPrototype) {
-            throw new Error(`Conflicting bindings of ${namePrototype}`);
-          }
-          // eslint-disable-next-line no-continue
-          continue;
-        }
-        intrinsics[namePrototype] = intrinsicPrototype;
+  const addIntrinsics = newIntrinsics => {
```

Comment:
The change below here is merely changing the intrinsicsCollector to a bag of closure functions instead of using concise methods. They were already used as bare closure functions, but this needed to be more explicit to avoid dynamic dispatch.

This is a case where dynamic dispatch would have been just as safe, but it was as easy to refactor this way and avoid lint-rule comment noise.

Reply:
Nice!